aca_hi_bgd/update_bgd_events.py

```diff
@@ -47,548 +72,1173 @@ def get_opt(args=None):
-            aca_l0.get_slot_data(start.secs - 20, stop.secs + 20, slot=slot).data
-        )
-        slotdata[slot]["SLOT"] = slot
+        if len(guide_cat) == 0:
+            continue
+        guide_slots = guide_cat[guide_cat["slot"] == slot]
+        if len(guide_slots) != 1:
+            continue
+        guide_slot = guide_slots[0]
+        if np.in1d(guide_slot["type"], ["BOT", "GUI"]):
+            # Use the full current agasc for these lookups instead of the commanded magnitude
+            star = agasc.get_star(guide_slot["id"], agasc_file="agasc*")
+            slot_mag[slot] = star["MAG_ACA"]
+        else:
+            slot_mag[slot] = guide_slot["mag"]
+    return slot_mag
+
+
+def get_slots_metrics(slots_data: dict) -> dict:
+    """
+    Get metrics for each slot
+
+    Parameters
+    ----------
+    slots_data : dict
+        A dictionary of slot data tables with keys 0-7 and tables with columns
+        "TIME", "IMGFUNC", "IMGSIZE", "outer_min_7_magsub", "bgdavg_es", "threshold"
+
+    Returns
+    -------
+    slots_metrics : dict
+        Dictionary of metrics for each slot
+    """
+    slots_metrics = {}
+    for slot in range(8):
+        if slot not in slots_data:
+            continue
+        slot_data = slots_data[slot]
+        ok = slot_data["IMGFUNC"] == 1
+        if np.count_nonzero(ok) == 0:
+            continue
+        if np.median(slot_data["IMGSIZE"][ok]) == 6:
+            ok6 = (slot_data["IMGFUNC"] == 1) & (slot_data["IMGSIZE"] == 6)
+            slots_metrics[slot] = {
+                "bg_col": "bgdavg_es",
+                "threshold": np.median(slot_data["threshold"][ok6]),
+                "max": slot_data["bgdavg_es"][ok6].max(),
+            }
+        elif np.median(slot_data["IMGSIZE"][ok]) == 8:
+            ok8 = (slot_data["IMGFUNC"] == 1) & (slot_data["IMGSIZE"] == 8)
+            slots_metrics[slot] = {
+                "bg_col": "outer_min_7_magsub",
+                "max": slot_data["outer_min_7_magsub"][ok8].max(),
+                "threshold": np.median(slot_data["threshold"][ok8]),
+            }
+        else:
+            continue
+    return slots_metrics
+
+
+def get_manvr_events(start: CxoTimeLike, stop: CxoTimeLike, obsid: int) -> tuple:
+    """
+    Review a single dwell for high background events.
+
+    Parameters
+    ----------
+    start : CxoTimeLike
+        Start of interval to process
+    stop : CxoTimeLike
+        End of interval to process
+    obsid : int
+        Observation ID
+
+    Returns
+    -------
+    tuple (bgd_events, dwell_stop, slots_metrics)
+        bgd_events : astropy.table.Table
+            Table of high background events
+        dwell_stop : CxoTime
+            Stop time of the last processed interval
+        slots_metrics : dict
+            Dictionary of metrics for each slot
+
+    """
+    slot_mag = get_slot_mags(start)
+
+    bgd_events = []
+
+    try:
+        sd_table = get_aca_images(start, stop, source="cxc", bgsub=True)
```

Comment:
Actually this will need a bit of a tweak once we get kadi events from maude.  I think we'll then want to decide on a per-interval basis which data source makes sense (get_aca_images doesn't have that convenience baked in as far as I remember).

Reply:
I mean, this code will be fine either way (either just having it fail when "cxc" data is unavailable in the range, or updating it to work from maude data in the just-maude range) but it is a good point about kinds of trending failures over newly available events.